aws/resource_aws_api_gateway_documentation_version_test.go

```diff
@@ -0,0 +1,266 @@
+					resource.TestCheckResourceAttrSet(resourceName, "rest_api_id"),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAWSAPIGatewayDocumentationVersionExists(n string, res *apigateway.DocumentationVersion) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No API Gateway Documentation Version ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+		apiId, version, err := decodeApiGatewayDocVersionId(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		req := &apigateway.GetDocumentationVersionInput{
+			DocumentationVersion: aws.String(version),
+			RestApiId:            aws.String(apiId),
+		}
+		docVersion, err := conn.GetDocumentationVersion(req)
+		if err != nil {
+			return err
+		}
+
+		*res = *docVersion
+
+		return nil
+	}
+}
+
+func testAccCheckAWSAPIGatewayDocumentationVersionDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_api_gateway_documentation_version" {
+			continue
+		}
+
+		version, apiId, err := decodeApiGatewayDocVersionId(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		req := &apigateway.GetDocumentationVersionInput{
+			DocumentationVersion: aws.String(version),
+			RestApiId:            aws.String(apiId),
+		}
+		_, err = conn.GetDocumentationVersion(req)
+		if err != nil {
+			if isAWSErr(err, apigateway.ErrCodeNotFoundException, "") {
+				return nil
+			}
+			return err
+		}
+
+		return fmt.Errorf("API Gateway Documentation Version %q still exists.", rs.Primary.ID)
+	}
+	return nil
+}
+
+func testAccAWSAPIGatewayDocumentationVersionBasicConfig(version, apiName string) string {
+	return fmt.Sprintf(`
+resource "aws_api_gateway_documentation_version" "test" {
+  version = "%s"
+  rest_api_id = "${aws_api_gateway_rest_api.test.id}"
+  depends_on = ["aws_api_gateway_documentation_part.test"]
+}
+
+resource "aws_api_gateway_documentation_part" "test" {
+  location {
+    type = "API"
+  }
+  properties = "{\"description\":\"Terraform Acceptance Test\"}"
+  rest_api_id = "${aws_api_gateway_rest_api.test.id}"
+}
+
+resource "aws_api_gateway_rest_api" "test" {
+  name = "%s"
+}
+`, version, apiName)
+}
+
+func testAccAWSAPIGatewayDocumentationVersionAllFieldsConfig(version, apiName, stageName, description string) string {
+	return fmt.Sprintf(`
+resource "aws_api_gateway_documentation_version" "test" {
+  version = "%s"
+  rest_api_id = "${aws_api_gateway_rest_api.test.id}"
+  description = "%s"
+  depends_on = ["aws_api_gateway_documentation_part.test"]
+}
+
+resource "aws_api_gateway_documentation_part" "test" {
```

Comment:
Nitpick: Can we reuse something from the `aws_api_gateway_documentation_part` test configurations instead of duplicating this whole large configuration? 😄

Reply:
I'm not sure it's a good idea as tests should ideally be as isolated as possible, at least cross resource, IMO.